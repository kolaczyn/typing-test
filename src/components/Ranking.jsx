import React from 'react'

export default function Ranking({ scores }) {
   return scores.length ? (
    <div className="ranking element">
      <table>
        <thead>
          <tr>
            <td className="ranking__place"></td>
            <td className="ranking__wpm">WPM</td>
            <td className="ranking__accuracy">Accuracy</td>
            <td className="ranking__time">Ago</td>
          </tr>
        </thead>
        <tbody>
          {scores.map((score, idx) =>
            <tr key={idx}>
              <td>{`${idx + 1}.`}</td>
              <td>{score.wpm}</td>
              <td>{score.accuracy}</td>
              <td>{score.timeAgo}</td>
            </tr>
          )}
        </tbody>
      </table>
    </div>
  ): null;
}
