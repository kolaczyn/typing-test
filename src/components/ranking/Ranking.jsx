import React from 'react';

import {
  rankingData,
  rankingDataLabels,
} from '../../static/fixtures/rankingData';
import Table from '../common/table';
import Box from '../common/box';
import Container from '../common/Container';

export default function Ranking() {
  return (
    <Container>
      <Box title='Ranking'>
        <Table labels={rankingDataLabels} data={rankingData} />
      </Box>
    </Container>
  );
}
