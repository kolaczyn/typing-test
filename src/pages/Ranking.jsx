import React from 'react'

import Box from '../components/ui/Box';

export default function Ranking() {
  return (
    <Box>Welcome to ranking!</Box>
  )
}
