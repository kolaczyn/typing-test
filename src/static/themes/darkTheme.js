import { DARK_THEME } from '../../reducers/theme/actions';

const darkTheme = {
  palette: {
    box: '#222',
    border: '#1c1e21',
    background: '#303030',
    text: '#fff',

    green: '#55efc4',
    blue: '#00a1e4',
    red: '#e17055',
    yellow: '#e8e9a1',
    lightGray: 'bisque',
  },
  themeName: DARK_THEME,
};

export default darkTheme;
