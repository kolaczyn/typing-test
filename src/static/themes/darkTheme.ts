import Action from '../../reducers/theme/actions';
import ThemeType from './ThemeType';

// https://flatuicolors.com/palette/defo
const darkTheme: ThemeType = {
  palette: {
    box: '#2c3e50',
    border: '#27394B',
    background: '#34495e',
    text: '#ecf0f1',

    green: '#2ecc71',
    blue: '#3498db',
    red: '#f96454',
    yellow: '#f1c40f',
    lightGray: '#bdc3c7',
    darkGray: '#bdc3c7',
  },
  themeName: Action.DARK_THEME,
};

export default darkTheme;
