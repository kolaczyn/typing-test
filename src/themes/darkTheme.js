export default {
  palette: {
    box: '#222',
    background: '#303030',
    lightGray: 'bisque',
    green: '#55efc4',
    red: '#e17055',
    text: '#fff',
  },
  containerWidth: '768px',
};
