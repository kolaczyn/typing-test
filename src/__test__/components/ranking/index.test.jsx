import React from 'react';
import { ThemeProvider } from 'styled-components';
import { cleanup, render } from '@testing-library/react';

import Ranking from '../../../components/ranking';
import darkTheme from '../../../static/themes/darkTheme';

afterEach(cleanup);

describe('<Ranking />', () => {
  it('renders <Ranking />', () => {
    const { container } = render(
      <ThemeProvider theme={darkTheme}>
        <Ranking />
      </ThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
