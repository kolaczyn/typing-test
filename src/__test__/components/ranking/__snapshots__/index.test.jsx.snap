// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ranking /> renders <Ranking /> 1`] = `
<section
  class="styles__Container-sc-9qjjy7-0 dsJXrG"
  title="7"
>
  <header
    class="styles__Header-sc-9qjjy7-1 ejqAhC"
  >
    Ranking
  </header>
  <table
    class="styles-sc-125e0ud-0 bxUohM"
  >
    <thead
      class="styles__Thead-sc-125e0ud-1 eYSPVw"
    >
      <tr>
        <td />
        <td>
          Username
        </td>
        <td>
          WPM
        </td>
        <td>
          When
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1.
        </td>
        <td>
          Satania
        </td>
        <td>
          120
        </td>
        <td>
          1h
        </td>
      </tr>
      <tr>
        <td>
          2.
        </td>
        <td>
          Gabriel
        </td>
        <td>
          89
        </td>
        <td>
          3h
        </td>
      </tr>
      <tr>
        <td>
          3.
        </td>
        <td>
          Vigne
        </td>
        <td>
          70
        </td>
        <td>
          5h
        </td>
      </tr>
      <tr>
        <td>
          4.
        </td>
        <td>
          Raphi
        </td>
        <td>
          68
        </td>
        <td>
          5d
        </td>
      </tr>
      <tr>
        <td>
          5.
        </td>
        <td>
          Megumin
        </td>
        <td>
          59
        </td>
        <td>
          2d
        </td>
      </tr>
      <tr>
        <td>
          6.
        </td>
        <td>
          Aqua
        </td>
        <td>
          55
        </td>
        <td>
          4d
        </td>
      </tr>
      <tr>
        <td>
          7.
        </td>
        <td>
          Darkness
        </td>
        <td>
          50
        </td>
        <td>
          1h
        </td>
      </tr>
      <tr>
        <td>
          8.
        </td>
        <td>
          Emilia
        </td>
        <td>
          30
        </td>
        <td>
          30m
        </td>
      </tr>
      <tr>
        <td>
          9.
        </td>
        <td>
          Rem
        </td>
        <td>
          23
        </td>
        <td>
          10h
        </td>
      </tr>
      <tr>
        <td>
          10.
        </td>
        <td>
          Ram
        </td>
        <td>
          10
        </td>
        <td>
          9h
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;
